### PR TITLE
Fire - Anna

### PR DESCRIPTION
**Pull request by annakim93**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails |  The model allows us to create and/or interact with a database; there are a number of methods that we inherit from ApplicationRecord (which was enough in this case s.t. we didn't need to define other db relations or custom methods)
Describe in your own words what the Controller is doing in Rails | The controller interacts with the model to get, update, delete, etc (CRUD) data and sends that data along with the appropriate format to render a view.
Describe in your own words what the View is doing in Rails | The view provides the view for the user to see (renders the page on the browser)
Describe an edge-case controller test you wrote | Wrote tests to check for redirection when an action was requested to be completed on an invalid task
What is the purpose of using strong params? (i.e. the params method in the controller) | strong params determine which parameter items in params we have access to; allows for a shortcut in other methods that need to access params
How are Rails migrations related to Rails models? | Rails migrations are the steps that we take to get to a working database (create the db; add a col we need; rename a col, etc)
Describe one area of Rails that are still unclear on | There's a lot under the hood that is unclear and being taken for granted, but it gets clearer with more practice!

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
+class TasksController < ApplicationController
+  # Helper Methods
+  def task_error_notice
+    flash[:notice] = "oops! that task does not exist..."
```
Comment (jmaddox19) [inline app/controllers/tasks_controller.rb:4]: You're using flash messages already?? You're so ahead of the game! 😮 👏

**Comment by jmaddox19** [comment]
TaskList Feedback

# Task List




## Major Learning Goals/Code Review





| Criteria                                           | yes/no, and optionally any details/lines of code to reference |
| -------------------------------------------------- | ------------------------------------------------------------- |
| At least 6 commits with meaningful commit messages | ✔️    Yes! So many commits with so many meaningful messages!                                                      |
| Routes follow RESTful conventions                  | ✔️                                                            |
| Uses named routes (like `_path`)                   | ✔️                                                            |
| Creates Models and migrations                      | ✔️                                                            |
| Creates styled views                               | No styling                                                         |
| Handles errors like nonexistant tasks              | ✔️                                                            |
| Uses `form_with` to render forms in Rails          | ✔️                                                            |

## Functional Requirements/Manual Testing

| Functional Requirement                                                         | yes/no |
| ------------------------------------------------------------------------------ | ------ |
| Successfully handles index & show                                              | ✔️     |
| index & show tests pass                                                        | ✔️     |
| Successfully handles: New, Create                                              | ✔️     |
| New, Create tests pass                                                         | ✔️     |
| Successfully handles: Edit, Update                                             | ✔️     |
| Edit, Update tests pass with valid & invalid task ids                          | ✔️    |
| Successfully handles: Destroy, Task Complete                                   | ✔️     |
| Tests for Destroy & Task Complete include tests for valid and invalid task ids | Yes for destroy but not for Task Complete     |

## Overall Feedback

Great work! You made a fully-functional multi-page Ruby on Rails website!!! In just a week! And it looks SO STINKING GOOD! 

And all the code is easy to read and understand! 👏👏👏

| Overall Feedback                | Criteria                                                                                                                                                                                      | yes/no |
| ------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------ |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 6+ in Functional Requirements                                                                                                                                            |✔️
| Yellow (Approaches Standards)   | 3+ in Code Review && 5+ in Functional Requirements, or the instructor judges that this project needs special attention                                                                        |
| Red (Not at Standard)           | 0-2 in Code Review or 0-4 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |





## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality              | Yes? |
| -------------------- | ---- |
| Perfect Indentation  | ✅    |
| Elegant/Clever       | ✅    |
| Descriptive/Readable | ✅    |
| Concise              | ✅    |
| Logical/Organized    | ✅    |
